src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/DeferredProjectWorkspaceService.cs

```diff
@@ -76,5 +76,26 @@ internal sealed class DeferredProjectWorkspaceService : IDeferredProjectWorkspac
                     projectFilePath, projectProperties: projectProperties);
             }
         }
+
+        public async Task<string> GetProjectTypeGuidAsync(string projectFilePath)
+        {
+            var workspace = SolutionWorkspaceService.CurrentWorkspace;
+            var solutionPath = SolutionWorkspaceService.SolutionFile;
+            var indexService = workspace.GetIndexWorkspaceService();
+            var indexedSolutionProjectTypes = (await indexService.GetFileDataValuesAsync<ProjectBaseTypesInSolution>(solutionPath,
```

Comment:
nit: I would assign the awaited value and then use it, but it would be the same result